### ISSUE TITLE
sql: error decoding bytes: couldn't decode type unknown

### ISSUE THREAD

**Issue by jordanlewis** [cockroachdb/cockroach#34524]
```
root@127.0.0.1:62165/defaultdb> create table a (a int primary key);
CREATE TABLE

Time: 5.724ms

root@127.0.0.1:62165/defaultdb> (select null from a) except (values((select 1 from a limit 1)), (1));
pq: error decoding 2 bytes: couldn't decode type unknown
```

**Comment by yuzefovich** [comment]
Turning off `distsql` removes this error. I'm not sure what exactly changes when we turn off `distsql`.

**Comment by jordanlewis** [comment]
It changes whether or not we use a wrapped, local values node (distsql=off) or a distributed, distsql values node (distsql=on). So there's some kind of bug in the type propagation in the distsql values node.

**Comment by jordanlewis** [comment]
@yuzefovich and I did a deep dive on this today. We determined that there's two causes to the problem. One is that the created plan has an `unknown` output type for one of the sides of the join. The other is that the special casing done in the execution component of union does not have sufficient smarts to deal with this case. Sometimes, if there's a plan with `unknown` output type, execution will work, but we can't always make it work - the output type information for both sides really needs to be correct.

We discussed some solutions with @rytaft. It seems like the best way for now to remove the `unknown` output type plan child is to create a normalization rule that propagates the type of a set operation to its children if either is unknown, so assigning this to her to take a look.
